recipe/session/config_test.go

```diff
@@ -1252,3 +1254,66 @@ func TestSuperTokensInitWithDefaultAPIGateWayPathandCustomAPIBasePath(t *testing
 	}
 	assert.Equal(t, sp.AppInfo.APIBasePath.GetAsStringDangerous(), "/hello")
 }
+
+func TestInvalidSameSiteConfig1(t *testing.T) {
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			AppName:       "SuperTokens",
+			WebsiteDomain: "http://localhost:3000",
+			APIDomain:     "http://api.supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			Init(&sessmodels.TypeInput{}),
+		},
+	}
+
+	BeforeEach()
+	defer AfterEach()
+	err := supertokens.Init(configValue)
+	assert.Equal(t, err.Error(), "Since your API and website domain are different, for sessions to work, please use https on your apiDomain and dont set cookieSecure to false.")
+}
+
+func TestInvalidSameSiteConfig2(t *testing.T) {
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			AppName:       "SuperTokens",
+			WebsiteDomain: "http://supertokens.io",
+			APIDomain:     "http://localhost:8000",
+		},
+		RecipeList: []supertokens.Recipe{
+			Init(&sessmodels.TypeInput{}),
+		},
+	}
+
+	BeforeEach()
+	defer AfterEach()
+	err := supertokens.Init(configValue)
+	assert.Equal(t, err.Error(), "Since your API and website domain are different, for sessions to work, please use https on your apiDomain and dont set cookieSecure to false.")
+}
+
+func TestInvalidSameSiteConfig3(t *testing.T) {
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			AppName:       "SuperTokens",
+			WebsiteDomain: "http://supertokens.io",
+			APIDomain:     "http://supertokensapi.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			Init(&sessmodels.TypeInput{}),
+		},
+	}
+
```

Comment:
Add the following tests if they don't already exist:
- When using localhost or IP for both apiDomain and websiteDomain, adding sameSite none should be allowed
- When apiDomain and websiteDomain are both an IP or localhost with https, then none should be allowed
- When apiDomain and websiteDomain are both non localhost and non IP, but with https, then none should be allowed.
- When apiDomain is non localhost and non IP (with https), websiteDomain is http://localhost:3000, then none should be allowed.

Please make sure all the possible inputs are covered - this is a very important config.